### DISCUSSION TITLE
Templating yaml keys

### DISCUSSION THREAD

**Discussion by miro-test** [k14s/ytt#560] [General]
Is it possible to paste values to YAML keys?

i need to genererate gitlab pipelines with different job names.
something like this
```
#@ data.values.ENV+":plan":
  stage: plan
  ....

#@ data.values.ENV+"deploy":
  stage: deploy
  ...
```
schema file
```
#@data/values-schema
---
ENV: dev
```

**Comment by vmunishwar** [comment]
Yes, ytt provides a feature called '[Text Templating](https://carvel.dev/ytt/docs/latest/ytt-text-templating/)' that allows setting up YAML map keys dynamically with different data values. Additional information can be found in this [FAQ](https://carvel.dev/ytt/docs/latest/faq/#how-can-i-dynamically-set-or-replace-map-key-as-a-data-value-in-my-template)
